File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        // while(s[i]!='\0' && s[i]!='\n'){

        // }
        string ans="";
        int len=0;
        for(int i=s.length()-1;i>=0;i--){
            // if('a'<=s[i] && 'z'>=s[i]){
                if(isalpha(s[i]))len++;
            // }else if('a'<=s[i+1] && 'z'>=s[i+1]){
                else if(len>0){
                ans+=s.substr(i+1,len);
                ans+=" ";
                len=0;
            }
        }

        if(len > 0) {
            ans += s.substr(0, len);
        }

        while(!ans.empty() && ans.back() == ' ') {
                    ans.pop_back();
                }
        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        arr=s.split()
        print(arr)
        arr=arr[::-1]
        st=""
        n=len(arr)
        j=0
        for i in arr:
            # st.append(i)
            # st.append(" ")
            st+=i
            if j<n-1:
                st+=" "
            j+=1
        return st
        